#### Breast Tumour prediction 
- ![Data Source](https://archive.ics.uci.edu/dataset/16/breast+cancer+wisconsin+prognostic)

- predicting whether the tumour is benign or malignant

***Import Libraries***

In [39]:
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import confusion_matrix


***Importing dataset***

In [ ]:
dataset = pd.read_csv("../dataFiles/breast_cancer.csv")
x = dataset.iloc[:, 1:-1].values
y = dataset.iloc[:, -1]

print(f'{x}\n{y}')

***Splitting the dataset into Traning and Test Set***

In [41]:
x_train, x_test, y_train, y_test = train_test_split(x,y, test_size=0.20, random_state = 0)

***Training the logistic Regression model on the Training Set***

In [42]:
classifier = LogisticRegression(random_state=0)
classifier.fit(x_train, y_train) # pass independent and dependent train variables

LogisticRegression(random_state=0)

***Predicting test Set results***

In [43]:
y_pred = classifier.predict(x_test) 


***Predicting the test set results***

In [44]:

print(classifier.predict(([[5,1,1,1,2,1,3,1,1]])))

[2]


***Making the Confusion Matrix***

- shows how accurate the model was at predicting

In [45]:
cm = confusion_matrix(y_test, y_pred)
print(cm)

[[84  3]
 [ 3 47]]
Accuracy score: 131


84 - correct predictions that a tumour is benign
3 - incorrect predictions that a tumour is malignant - false negative
47 - correct predictions that a tumour is malignant
3  - incorrect predictions that a tumour is benign - false positive

***Computing the accuracy with K-fold Cross validation***

- Create 10 accuracies to check that model performs well on other unknow data
- Accuracy is the number of correct predictions divided by the number of observations

In [46]:
#calculating accuracy manually on the above data
accuracy = (84+47)/(3+3+84+47)
print(accuracy)

0.9562043795620438


In [48]:
from sklearn.model_selection import cross_val_score
accuracies = cross_val_score(estimator=classifier, X=x_train, y=y_train, cv=10) # dafault no of folds is 10 and the 

#print average of accuracies
print (f'Accuracy: {accuracies.mean()*100:.2f}%')
# Print standard deviation of the accuracies
print (f'Standard Deviation: {accuracies.std()*100:.2f}%')

Accuracy: 96.70%
Standard Deviation: 1.97%


The standard deviation means that the range of accuracies fall between accuracy +- 1.97